### PR TITLE
AG-11680 Area series line style

### PR DESCRIPTION
**Pull request by jacobp100**
https://ag-grid.atlassian.net/browse/AG-11680

### CODE REVIEW

File: packages/ag-charts-community/src/chart/cartesianChart.test.ts [hunk 1]
```diff
@@ -240,7 +241,9 @@ describe('CartesianChart', () => {
         it.each(YKEYS)(`should render series with yKey [%s] appropriately`, async (yKey) => {
             const { chart: testee, highlightManager, nodeData } = await buildChart({ ...tcOptions }, yKey);
             highlightManager.updateHighlight(testee.id, nodeData?.nodeData[3]);
-            await compare(testee);
+            await compare(testee, {
+                failureThreshold: 2,
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/cartesianChart.test.ts:245]: These threshold changes shouldn't be needed, let me sync the branch and see what might be causing you to need this.

File: packages/ag-charts-community/src/chart/series/cartesian/areaSeries.ts [hunk 2]
```diff
@@ -504,42 +577,38 @@ export class AreaSeries extends CartesianSeries<
...
-        }
-        if (lastPoint) {
-            fillPath.lineTo(lastPoint.x, lastPoint.y);
-        }
-        fillPath.closePath();
+        const { line } = this.properties;
+
+        let start = 0;
+        fill.path.clear(true);
+        splitPartialPaths(points.map((point) => point.point)).forEach((topPath) => {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/cartesian/areaSeries.ts]: I'd have a small concern about performance regression here, as `for ()` loops and iterators seems to scale much better than `.map()` and `.forEach()` - could you validate that for large-scale data-sets (e.g. maybe try switching one of these series in this example to `area`?)
https://charts-staging.ag-grid.com/javascript/benchmarks/#large-scale-dataset-test-4x-100k-series

File: packages/ag-charts-community/src/chart/series/cartesian/areaSeries.ts [hunk 3]
```diff
@@ -504,42 +577,38 @@ export class AreaSeries extends CartesianSeries<
...
+
         strokePath.clear(true);
-        for (const { point } of strokeData.points) {
-            if (point.moveTo) {
-                strokePath.moveTo(point.x, point.y);
-            } else {
-                strokePath.lineTo(point.x, point.y);
-            }
-        }
+        splitPartialPaths(strokeData.points.map((point) => point.point)).forEach((path) => {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/cartesian/areaSeries.ts]: Ditto

File: packages/ag-charts-community/src/chart/series/cartesian/areaSeries.ts (hunk 2 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/series/cartesian/areaSeries.ts]: Updated. We still have array construction with `splitPairData` - but that's only used for animation. Assume that's ok?

File: packages/ag-charts-community/src/chart/series/cartesian/areaSeries.ts (hunk 3 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/series/cartesian/areaSeries.ts]: Updated

File: packages/ag-charts-community/src/chart/cartesianChart.test.ts (hunk 1 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/cartesianChart.test.ts:245]: Updated these to ensure it's a pixel threshold - it's only 2 pixels. Assume that's ok?

**Pull request merged**